webpack.config.js

```diff
@@ -21,46 +20,204 @@ module.exports = {
+            // overflows (https://github.com/webpack/webpack/issues/1721), and
+            // there is no need for webpack to parse them - they can just be
+            // included as-is.
+            /highlight\.js[\\\/]lib[\\\/]languages/,
+
+            // olm takes ages for webpack to process, and it's already heavily
+            // optimised, so there is little to gain by us uglifying it.
+            /olm[\\\/](javascript[\\\/])?olm\.js$/,
+        ],
         rules: [
-            { enforce: 'pre', test: /\.js$/, use: "source-map-loader", exclude: /node_modules/, },
-            { test: /\.js$/, use: "babel-loader", include: path.resolve(__dirname, 'src') },
             {
-                test: /\.wasm$/,
-                loader: "file-loader",
-                type: "javascript/auto", // https://github.com/webpack/webpack/issues/6725
+                test: /\.(ts|js)x?$/,
+                exclude: /node_modules/,
+                loader: 'babel-loader',
                 options: {
-                    name: '[name].[hash:7].[ext]',
-                    outputPath: '.',
-                },
+                    // We have the babel config here rather than a file to help
+                    // make it clearer what we're compiling to.
+                    cacheDirectory: true,
+                    sourceMaps: true,
+                    presets: [
+                        ["@babel/preset-env", {
+                            "targets": {
+                                "browsers": [
+                                    "last 2 versions",
+                                ]
+                            },
+                        }],
+                        "@babel/preset-typescript",
+                        "@babel/preset-flow",
+                        "@babel/preset-react",
+                    ],
+                    plugins: [
+                        // Most of these plugins are for the react-sdk to make itself
+                        // work correctly.
+                        ["@babel/plugin-proposal-decorators", {"legacy": true}],
+                        "@babel/plugin-proposal-export-default-from",
+                        "@babel/plugin-proposal-numeric-separator",
+                        "@babel/plugin-proposal-class-properties",
+                        "@babel/plugin-proposal-object-rest-spread",
+                        "@babel/plugin-transform-flow-comments",
+                        "@babel/plugin-syntax-dynamic-import",
+                        "@babel/plugin-transform-runtime"
+                    ],
+                }
+            },
+            {
+                test: /\.css$/,
+                use: [
+                    MiniCssExtractPlugin.loader,
+                    {
+                        loader: 'css-loader',
+                        options: {
+                            importLoaders: 1,
+                            sourceMap: true,
```

Comment:
also surprising, although presumably this is a different ball game since this is webpack plugins generating output webpack doesn't actually know about

Reply:
like above, the sourcemaps are wrong through webpack alone.